### PR TITLE
Migration script fixes for team refactoring

### PR DESCRIPTION
**Pull request by Half-Shot**
This tweaks the migration script so it *should* migrate the old format stores into the new format with separated team tokens from rooms and all that fancy stuff.

This script has been tested to ensure it works with the 0.3.2 data format.

### CODE REVIEW

File: src/SlackClientFactory.ts [hunk 1]
```diff
@@ -140,14 +140,16 @@ export class SlackClientFactory {
...
-        const opts = this.config.slack_client_opts;
+        const opts = this.config ? this.config.slack_client_opts : undefined;
         const slackClient = new WebClient(token, {
             logger: {
                 setLevel: () => {}, // We don't care about these.
                 setName: () => {},
                 debug: (msg: string) => {
                     // non-ideal way to detect calls to slack.
-                    const match = /apiCall\('([\w\.]+)'\) start/.exec(msg);
+                    webLog.debug.bind(webLog);
```
Comment (turt2live) [inline src/SlackClientFactory.ts]: leftover by accident?

File: src/scripts/migrateToPostgres.ts [hunk 2]
```diff
@@ -67,62 +73,131 @@ async function main() {
...
+
+    const preTeamMigrations = () => Promise.all(allRooms.map(async (room, i) => {
+        // This is an old format remote
+        // tslint:disable-next-line: no-any
+        const at = (room.remote as any).access_token;
+        if (!at) {
+            return;
+        }
+        try {
+            if (at) {
```
Comment (turt2live) [inline src/scripts/migrateToPostgres.ts]: usually the if comes before the try unless the if has multiple branches

File: src/scripts/migrateToPostgres.ts [hunk 3]
```diff
@@ -67,62 +73,131 @@ async function main() {
...
 
-    const slackUserMigrations = allSlackUsers.map(async (user, i) => {
+    const slackUserMigrations = () => Promise.all(allSlackUsers.map(async (user, i) => {
         // tslint:disable-next-line: no-any
-        const ghost = SlackGhost.fromEntry(null as any, user, null);
-        await pgres.upsertUser(ghost);
+        let ghost = SlackGhost.fromEntry(null as any, user, null);
+        if (!ghost.slackId || !ghost.teamId) {
+            const localpart = ghost.userId.split(":")[0];
+            // TODO: we are making an assumption here that the prefix ends with _
```
Comment (turt2live) [inline src/scripts/migrateToPostgres.ts]:
```suggestion
            // XXX: we are making an assumption here that the prefix ends with _
```

File: src/scripts/migrateToPostgres.ts [hunk 4]
```diff
@@ -67,62 +73,131 @@ async function main() {
...
-        );
-        log.info("Completed migration");
-    } catch (ex) {
-        log.error("An error occured while migrating databases:");
-        log.error(ex);
-        log.error("Your existing databases have not been modified, but you may need to drop the postgres table and start over");
-    }
+    }));
+    await eventMigrations();
+    log.info("Finished eventMigrations");
```
Comment (turt2live) [inline src/scripts/migrateToPostgres.ts]: Would be good to pair these with "Starting eventMigrations" and such, to show progress.

File: src/SlackClientFactory.ts (hunk 1 above)
Comment (Half-Shot) [reply src/SlackClientFactory.ts]: Nope, intentional.

**Pull request merged**